src/azure-cli-core/azure/cli/core/tooling.py

```diff
@@ -0,0 +1,194 @@
+"""tooling integration"""
```

Comment:
`tooling.py` is production code. The reason I put it in the CLI is that this code relies on a lot of the CLI's details and having that on the CLI's side will allow for updating it together with the CLI, if it ever needs updating. Our tool, the VSCode extension, uses mostly `tooling.py` to access the CLI.